java/arcs/android/storage/database/DatabaseGarbageCollectionPeriodicTask.kt

```diff
@@ -28,6 +28,15 @@ class DatabaseGarbageCollectionPeriodicTask(
     private val log = TaggedLog { WORKER_TAG }
     init { log.debug { "Created." } }
 
+    // Note on `runBlocking` usage:
+    // `doWork` is run synchronously by the work manager.
+    // This is one of the rare cases that runBlocking was intended for: implementing the
```

Comment:
I think ideally we'd switch to CoroutineWorker and avoid this https://developer.android.com/topic/libraries/architecture/workmanager/advanced/coroutineworker
may be worth a todo here or filing a bug

Reply:
Oh, I'm glad you commented, it seems I didn't investigate this too deeply.

I think we might not even need need `CoroutineWorker`... we can just use the base `ListenableWorker` which returns a future. Should be an easy change, I'll make a follow-up task.